### PR TITLE
Heroku Organization App Improvements

### PR DESCRIPTION
**Pull request by phinze**
This is a rebase of #668 with an acceptance test and updates to make it conform to modern standards.

### CODE REVIEW

File: builtin/providers/heroku/resource_heroku_app.go [hunk 1]
```diff
@@ -9,13 +9,24 @@ import (
 	"github.com/hashicorp/terraform/helper/schema"
 )
 
+type herokuApplication struct {
```
Comment (catsby) [inline builtin/providers/heroku/resource_heroku_app.go]: Can we document what this is for? 
`application` "is used to store all the details of a heroku app" but now it has a child `App` that is a `herokuApplication` ?

Comment (phinze) [inline builtin/providers/heroku/resource_heroku_app.go]: Sure thing. On it.

Comment (phinze) [inline builtin/providers/heroku/resource_heroku_app.go]: Done in 45e29c3255d7b7d56d4c1e4e8334deb6645ed6df

**Comment by phinze** [comment]
Okay @catsby PTAL

**Comment by catsby** [comment]
LGTM

**Pull request merged**